server/app/controllers/DisplayableMessage.java

```diff
@@ -0,0 +1,39 @@
+package controllers;
+
+import views.components.ToastMessage.ToastType;
```

Comment:
At a high level, can you help me understand why we wouldn't convert our APIs to pass `ToastMessage`s around everywhere?

I definitely agree that the transition from a string to a strong type is an improvement, I'm just trying to figure out what `DisplayableMessage` is adding on top of `ToastMessage`.

Reply:
I think ToastMessage is more of a UX component, and DisplayableMessage is more of a data type. MVC patterns typically try to separate the responsibility between data and its visual representation.

ToastMessage contains properties relating to UX representation like duration and `canDismiss`, while DisplayableMessage is focused on the data only.

Additionally, DisplayableMessage allows us the flexibility to display messages using some other UX component apart from a ToastMessage if we need to at some point in the future.

If you think this is overengineered, I totally understand, and I can change it to just use ToastMessage throughout

Reply:
That makes sense to me. In this case, each controller ends up translating downstream calls which return errors to a DisplayableMessage and passing that to a view which then converts that to a Toast. In my opinion, our controllers are already tightly bound to the view itself, so it might not add a lot of value.

My main concern is that we may end up slowly having to add the concept of dismissable and the other view-related logic to DisplayableMessage over time.

I understand your rationale and I'll defer to you. You may want to consult @shanemc-goog or @bion for a second opinion.

Reply:
Requested approval from @bion to see what he thinks

Reply:
`DisplayableMessage` is pretty tightly bound to the toast concept -- the Severity `enum` refers directly to the toast UI component with its `toastType` field.

IMO this is a bit premature and we should factor out a common data class if/when there is a clear pattern for reuse across multiple components. See the [LocalizedStrings](https://github.com/civiform/civiform/blob/main/server/app/services/LocalizedStrings.java#L20) class for an example.

For further context, there are already other types of errors that we want to return to the user that do not fit the data scheme of `DisplayableMessage`, such as inline form field validation messages.

Reply:
Makes sense. I'll just go with using the ToastMessage directly then.